conans/client/api/conan_api.py

```diff
@@ -90,19 +167,37 @@ def user_update(self, user_name, user_pasword):
         return {}
 
     @api_method
-    def search_recipes(self, query, remote_patterns=None, local_cache=False):
-        remote = None
-        if remote_patterns is not None and len(remote_patterns) > 0:
-            remote = remote_patterns[0].replace("*", "remote")
+    def search_recipes(self, remotes, query):
+        search_recorder = SearchRecorder()
```

Comment:
The intermediate conversion to ``SearchRecorder`` doesn't make sense anymore, we need to improve this:

- Define the result data structure. It can directly be generic json if we want to, no problem.
- ``search.search_recipes`` can directly return the result data structure
- Maybe consider the error handling mechanism, what is an error and what not. I don't love attaching the .info field to the exception.